src/client/dfs/dfs.c

```diff
@@ -3314,14 +3319,26 @@ dfs_osetattr(dfs_t *dfs, dfs_obj_t *obj, struct stat *stbuf, int flags)
 		return EPERM;
 	if ((obj->flags & O_ACCMODE) == O_RDONLY)
 		return EPERM;
+	if (flags & DFS_SET_ATTR_MODE)
+		if ((stbuf->st_mode & S_IFMT) != (obj->mode & S_IFMT))
+			return EINVAL;
```

Comment:
This partially resolves [DAOS-5847](https://jira.hpdd.intel.com/browse/DAOS-5847), right?

Reply:
Yes.  I was not aware of that ticket.